aspnetcore/host-and-deploy/blazor/webassembly.md

```diff
@@ -394,3 +394,53 @@ A Blazor WebAssembly app can be initialized with the `loadBootResource` function
 External sources must return the required CORS headers for browsers to allow the cross-origin resource loading. CDNs usually provide the required headers by default.
 
 You only need to specify types for custom behaviors. Types not specified to `loadBootResource` are loaded by the framework per their default loading behaviors.
+
+## Rename DLL files to bypass proxies, firewalls, and antivirus software
+
+Some environments block the download of *.dll* files that are normally part of a Blazor app's published deployment. To address this issue, you can rename *.dll* files to use a different file extension and update the *blazor.boot.json* file.
+
+After publishing the app, use a shell script or DevOps build pipeline to rename *.dll* files to use a different file extension. Target the *.dll* files in the *wwwroot* directory of the app's published output (for example, *{CONTENT ROOT}/bin/Release/netstandard2.1/publish/wwwroot*).
+
+In the following examples, *.dll* files are renamed to use the *.bin* file extension.
+
+On Windows:
+
+```powershell
+dir .\_framework\_bin | rename-item -NewName { $_.name -replace ".dll\b",".bin" }
+((Get-Content .\_framework\blazor.boot.json -Raw) -replace '.dll"','.bin"') | Set-Content .\_framework\blazor.boot.json
+```
+
+On Linux or macOS:
+
+```console
+for f in _framework/_bin/*; do mv "$f" "`echo $f | sed -e 's/\.dll\b/.bin/g'`"; done
+sed -i 's/\.dll"/.bin"/g' _framework/blazor.boot.json
+```
+   
+To use a different file extension than *.bin*, replace *.bin* in the preceding commands.
+
+To address the compressed *blazor.boot.json.gz* and *blazor.boot.json.br* files, adopt either of the following approaches:
+
+* Remove the compressed *blazor.boot.json.gz* and *blazor.boot.json.br* files. Compression is disabled with this approach.
+* Recompress the updated *blazor.boot.json* file.
+
+The following Windows example uses a PowerShell script placed at the root of the project.
+
+*ChangeDLLExtensions.ps1:*:
+
+```powershell
+param([string]$filepath,[string]$tfm)
+dir $filepath\bin\Release\$tfm\wwwroot\_framework\_bin | rename-item -NewName { $_.name -replace ".dll\b",".bin" }
+((Get-Content $filepath\bin\Release\$tfm\wwwroot\_framework\blazor.boot.json -Raw) -replace '.dll"','.bin"') | Set-Content $filepath\bin\Release\$tfm\wwwroot\_framework\blazor.boot.json
+Remove-Item $filepath\bin\Release\$tfm\wwwroot\_framework\blazor.boot.json.gz
+```
+
+In the project file, the script is run after publishing the app:
+
+```xml
+<Target Name="ChangeDLLFileExtensions" AfterTargets="Publish" Condition="'$(Configuration)'=='Release'">
+  <Exec Command="powershell.exe -command &quot;&amp; { .\ChangeDLLExtensions.ps1 '$(SolutionDir)' '$(TargetFramework)'}&quot;" />
+</Target>
+```
+
+To provide feedback, visit [Consider changing URLs of .NET assemblies not to end with .dll (aspnetcore/issues #5477)](https://github.com/dotnet/aspnetcore/issues/5477).
```

Comment:
```suggestion
To provide feedback, visit [issue #5477 on GitHub](https://github.com/dotnet/aspnetcore/issues/5477).
```

... to avoid implying this is a general area for bikeshedding.